### PR TITLE
Prevent saving of map edits if there are no map features

### PR DESCRIPTION
**Pull request by mddilley**
## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/10991

This PR updates the edit component map flow to enable removing all features from a component and to disable the **Save Edit** button when that happens. The button enables again when another feature is added.

## Testing
**URL to test:** 

local or https://md-10991-prevent-save--atd-moped-main.netlify.app/moped/

**Steps to test:**
1. Create a new project and then add a new line or point component
2. Add some features and then save it
3. Edit the component that you just created and delete every feature by deselecting or using the trash can on drawn features
4. You should see the **Save Edit** button disable when you have no features left
5. Select a feature from the map or drawn one
6. The **Save Edit** button should be enabled again and you can save the update

---
#### Ship list
- [x] Code reviewed 
- [x] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js [hunk 1]
```diff
@@ -201,13 +201,10 @@ export default function TheMap({
             feature?.properties?.[sourceFeatureId] !== featureUniqueId // From CTN layers
         );
 
-        const isAtLeastOneFeatureRemaining = filteredFeatures.length > 0;
-        return isAtLeastOneFeatureRemaining
-          ? {
-              ...currentComponent,
-              [tableToInsert]: filteredFeatures,
-            }
-          : currentComponent;
```
Comment (mddilley) [inline moped-editor/src/views/projects/projectView/ProjectComponents/TheMap.js:210]: This was a temporary fix that prevented removing the last remaining feature from a component.

File: moped-editor/src/views/projects/projectView/ProjectComponents/index.js [hunk 2]
```diff
@@ -212,7 +213,7 @@ export default function MapView({ projectName, projectStatuses }) {
                   }
                   onSave={onSaveEditedComponent}
                   onCancel={onCancelComponentMapEdit}
-                  saveButtonDisabled={false}
+                  saveButtonDisabled={!doesDraftEditComponentHaveFeatures}
```
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectComponents/index.js:216]: The crux of it, /and/ it explains what's happening through your great name choice.

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/features.js [hunk 3]
```diff
@@ -173,3 +174,29 @@ export const useExistingDrawnFeatures = ({
     }
   }, [linkMode, draftEditComponent, drawControlsRef]);
 };
+
+/**
+ * Track whether a draft edit component has features or not
+ * @param {Object} draftEditComponent - the draft component storing feature edits
+ * @returns {Boolean} - does the draft component have features
+ */
+export const useDoesDraftEditComponentHaveFeatures = (draftEditComponent) => {
```
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/features.js:183]: Custom hook is perfect here! 💯

File: moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeFeatureCollections.js [hunk 4]
```diff
@@ -78,17 +100,15 @@ export const useAllComponentsFeatureCollection = (components) =>
         features: [],
       };
 
-    const allComponentfeatures = [];
+    const allComponentsFeatures = [];
```
Comment (frankhereford) [inline moped-editor/src/views/projects/projectView/ProjectComponents/utils/makeFeatureCollections.js:103]: Thanks for cleaning up the naming here too.

**Review by frankhereford — APPROVED** [review]
This does just exactly what it should do - wouldn't change a thing! 🚀 Thanks Mike!

**Review by johnclary — APPROVED** [review]
Looks great; works great—nice work @mddilley! 🚢 🚢 🚢 🚢 🚢 🚢 🚢 🚢

**Pull request merged**